taxonomy-editor-frontend/src/pages/editentry/ListTranslations.jsx

```diff
@@ -1,94 +1,351 @@
+                    })
+                    // Updated tags assigned for later use
+                    tagsToBeInserted = [...unDeletedTags].map(el => (el.tag));
+                }
+                else {
+                    newRenderedTranslations.push(allTagsObj)
+                }
+            })
+            setRenderedTranslations(newRenderedTranslations) // Set state
+        }
+        // Set state of main NodeObject
+        setNodeObject(prevState => {
+            const newNodeObject = {...prevState};
+            newNodeObject['tags_'+key] = tagsToBeInserted;
+            return newNodeObject
+        })
     }
 
     return ( 
-        <Box sx={{ml: 4}} className="translations">
+        <Box sx={{ml: 4}}>
             {/* Title */}
-            <Typography sx={{mt: 4, mb: 1}} variant='h5'>Translations</Typography>
+            <Stack direction="row" alignItems="center">
+                <Typography sx={{mt: 4, mb: 1}} variant='h5'>Translations</Typography>
+                <IconButton sx={{mt: 3.5, ml: 1, color: "#808080"}} onClick={handleOpen}>
+                    <AddBoxIcon />
+                </IconButton>
+            </Stack>
+
             {/* Main Language */}
-            <Typography variant='h6'>
-                { ISO6391.getName(nodeObject.main_language) }
-            </Typography>
+            <Stack direction="row" alignItems="center">
+                <Typography variant='h6'>
+                    { nodeObject && ISO6391.getName(nodeObject.main_language) }
+                </Typography>
+                <IconButton sx={{ml: 1, color: "#808080"}} onClick={() => handleAdd(nodeObject.main_language)}>
+                    <AddBoxIcon />
+                </IconButton>
+            </Stack>
+            
             {/* Render main language tags */}
             <Typography variant='h6'> 
-                {
-                    nodeObject["tags_"+nodeObject['main_language']].map((tag, index) => {
+                { nodeObject && 
+                    mainLang_renderedTranslations.map(({index, tag}) => {
                         return (
-                            // TODO: Key to be replaced by a UUID 
-                            <Paper key={index} component={Stack} direction="column" sx={{ml: 4, width: 200}}>
-                                <TextField 
-                                    size="small" 
-                                    sx={{mt: 1}} 
-                                    onChange = {event => {
-                                        changeData(nodeObject['main_language'], index, event.target.value)
-                                    }}
-                                    value={tag} 
-                                    variant="outlined" />
-                            </Paper>
+                            <Stack key={index} direction="row" alignItems="center">
+                                <Paper component={Stack} direction="column" sx={{ml: 4, width: 200}}>
```

Comment:
here you have a Stack inside another Stack. Are those needed? I wuold expect that TextField can be side by side with a button. Also is there any reason for that measure in the `width`?